install.sh

```diff
@@ -0,0 +1,140 @@
+#!/bin/bash
+
+# This script is meant to automate the deployment of Szurubooru via Docker/Docker-Compose,
+# and to automatically take the necessary steps to allow Szurubooru to run on ARM devices
+# such as the Raspberry Pi.
+
+###########################################[Notes]###########################################
+# * Written and tested on Raspberry Pi 4 running Raspbian 10 (armv7), heavily based on a fork
+#   by kaijuagenda for the Pi 3: https://github.com/kaijuagenda/szurubooru-rpi3
+#
+# * Modifies docker-compose.yml rather than relying on a static alternative file to hopefully
+#   avoid the need for parallel maintenance of two files.
+#############################################################################################
+
+echo "Welcome to Szurubooru.  This install script will ask you to"
+echo "set a few config values, then get szurubooru up and running."
+
+# Check to ensure Docker and Docker-Compose are both installed before proceeding
+if ! command -v docker &> /dev/null || ! command -v docker-compose &> /dev/null; then
```

Comment:
id -Gn | grep -i docker
^ will return 0 when the currently executing user in the docker group. You can unfortunately successfully run `docker` even if you don't have access to the socket it's talking to.

Additionally you could run `docker ps` to see if they can actually talk to their docker socket. It'll return 1 if they can't.

Reply:
Excellent information!

This section was only really meant to serve as an install check for Docker and Docker-Compose were installed, but I've added an additional check on `docker ps`'s  return status to ensure the current user can access the daemon/socket.  Thanks Jazzepi!

Reply:
No problem :) I love what you did with the script, so kudos on that.